pkg/wharfapi/provider.go

```diff
@@ -71,15 +72,40 @@ func (c Client) GetProvider(providerName string, urlStr string, uploadURLStr str
 	return providers[0], nil
 }
 
+func (c Client) PutProvider(provider Provider) (Provider, error) {
+	body, err := json.Marshal(provider)
+	if err != nil {
+		return Provider{}, err
+	}
+
+	log.WithField("provider", string(body)).Traceln()
+
+	apiURL := fmt.Sprintf("%s/api/provider", c.APIURL)
+	ioBody, err := doRequest("PUT | PROVIDER |", http.MethodPut, apiURL, body, c.AuthHeader)
+	if err != nil {
+		return Provider{}, err
+	}
+
+	defer (*ioBody).Close()
+
+	newProvider := Provider{}
```

Comment:
no need to initialize these values. More common to just declare them as zero values using the following syntax:

```suggestion
	var newProvider Provider
```